### PR TITLE
Documentation, test and normalization for cross entropy calculation

### LINKED ISSUE #47: Normalise predictions for cross entropy calculation
As discussed while implementing cross entropy in #38, we should add a normalisation of the predictions that probabilities add up to `1.0`:

> Adding a normalisation should be our best option.

_Originally posted by @cDenius in https://github.com/cirKITers/masKIT/pull/45#r705040666_

### PR DESCRIPTION
**Pull request by eileen-kuehn**
This PR extends documentation for the cross entropy calculation and adds normalisation as well as a simple unit test based on the discussion in [Stackoverflow](https://stackoverflow.com/a/47398312/10138546).

Closes #47.

### CODE REVIEW

**Comment by eileen-kuehn** [comment]
@cDenius, you can happily ignore that one failing test regarding project coverage :)

File: tests/test_utils.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import numpy as np
+
+from maskit.utils import cross_entropy
+
+
+def test_cross_entropy():
+    predictions = np.array([[0.25, 0.25, 0.25, 0.25], [0.01, 0.01, 0.01, 0.96]])
```
Comment (cDenius) [inline tests/test_utils.py:7]: The test case should be 

`predictions = np.array([[0.25, 0.25, 0.25, 0.25], [0.01, 0.01, 0.01, 0.97]])`

So the predictions add up to one. Then it is also identical to the output of the sklearn.metrics implementation: `0.7085330098933221` and not `0.71355817782`

**Review by cDenius — CHANGES_REQUESTED** [review]
I would change the test case so we are identical to the sklearn implementation

File: tests/test_utils.py (hunk 1 above)
Comment (eileen-kuehn) [reply tests/test_utils.py:7]: Hm, interesting comment.
This might actually change our use case as in the quantum case we will usually not add up to `1.0`. What is the expectation you have? Should we get the same result as with sklearn or not? So should we introduce a normalisation to always add up to the sum of target probabilities?

Comment (cDenius) [reply tests/test_utils.py:7]: Should we always add up to `1` in the quantum case as long as we have consider all of the `2^n` measured outputs? For the iris dataset with three classes, I would prefer if we would perform a softmax before calling the cross entropy.

Comment (cDenius) [reply tests/test_utils.py:7]: Not softmax, sorry. That doesn't make any sense. Adding a normalisation should be our best option.

**Review by cDenius — APPROVED** [review]
LGTM

**Review by maxfischer2781 — APPROVED** [review]
Looks good to me.
